ocp_utilities/operators.py

```
@@ -154,72 +158,88 @@ def install_operator(
-            ns.deploy(wait=True)
+        else:
+            ns = Namespace(client=admin_client, name=operator_namespace)
+            if not ns.exists:
+                ns.deploy(wait=True)
 
-    OperatorGroup(
-        client=admin_client,
-        name=name,
-        namespace=operator_namespace,
-        target_namespaces=target_namespaces,
-    ).deploy(wait=True)
+        OperatorGroup(
+            client=admin_client,
+            name=name,
+            namespace=operator_namespace,
+            target_namespaces=target_namespaces,
+        ).deploy(wait=True)
 
-    subscription = Subscription(
-        client=admin_client,
-        name=name,
-        namespace=operator_namespace,
-        channel=channel,
-        source=catalog_source.name if catalog_source else source,
-        source_namespace=operator_market_namespace,
-        install_plan_approval="Automatic",
-    )
-    subscription.deploy(wait=True)
-    wait_for_operator_install(
-        admin_client=admin_client,
-        subscription=subscription,
-        timeout=timeout,
-    )
+        subscription = Subscription(
+            client=admin_client,
+            name=name,
+            namespace=operator_namespace,
+            channel=channel,
+            source=catalog_source.name if catalog_source else source,
+            source_namespace=operator_market_namespace,
+            install_plan_approval="Automatic",
+        )
+        subscription.deploy(wait=True)
+        wait_for_operator_install(
+            admin_client=admin_client,
+            subscription=subscription,
+            timeout=timeout,
+        )
+    except Exception as ex:
+        LOGGER.error(f"{name} Install Failed. \n{ex}")
+        if must_gather_output_dir:
+            collect_must_gather(
+                must_gather_output_dir=must_gather_output_dir,
+                kubeconfig_path=kubeconfig,
+                cluster_name=cluster_name,
+                product_name=name,
+            )
```

you should raise the Exception in both cases (must_gather_output_dir or not)

---

@myakove updated the code